content/yaml-testing/detox-testing.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Detox (E2E) test automation
+description: How to run Detox tests with codemagic.yaml
+weight: 5
+
+---
+
+Detox testing refers to end-to-end (E2E) testing for React Native apps using the Detox framework. Detox provides a gray box testing solution, meaning it runs tests on a real device or emulator while also accessing the app's internal state for better synchronization.
+
+
+## Prerequisites
+
+1. A React Native project
+2. **npm/yarn** are pre-installed
+3. Xcode for iOS and Android Studio for Android are pre-installed
+4. **homebrew** is pre-installed
+
+{{<notebox>}}
+Pre-installed means that Codemagic machines already have the software ready to use, so there is no need to install it manually.
+{{</notebox>}}
+
+## Running Detox tests from codemagic.yaml
+
+1. Adjust your project's **package.json** file with Detox before starting Codemagic configurations:
+
+```json
+  "detox": {
+    "runnerConfig": "e2e/config.json",
+    "configurations": {
+      "ios.sim.release": {
+        "type": "ios.simulator",
+        "device": {
+          "type": "iPhone 14"
+        },
+        "binaryPath": "ios/build/Build/Products/Debug-iphonesimulator/MyReactNativeApp.app",
+        "build": "xcodebuild -workspace ios/YOUR_APP.xcworkspace -scheme YOUR_APP -configuration Debug -sdk iphonesimulator -derivedDataPath ios/build"
+
+      },
+    }
+  }
+```
+2. Install the required tools by running the following commands:
```

Comment:
"Running Detox tests in Codemagic"

We are going to add all the scripts we need to run to install additional software and execute Detox tests to the `scripts` section of codemagic.yaml.